Widgets/Widgets/PictureOfTheDayWidget.swift

```diff
@@ -0,0 +1,255 @@
+import WidgetKit
+import SwiftUI
+import WMF
+import UIKit
+
+// TODO: Move into `PictureOfTheDay+LocalizedStrings.swift`
+extension PictureOfTheDayWidget {
+
+    enum LocalizedStrings {
+        static let pictureOfTheDayWidgetTitle = WMFLocalizedString("potd-widget-title", value:"Picture of the day", comment: "Text for title of Picture of the day widget.")
+        static let pictureOfTheDayWidgetDescription = WMFLocalizedString("potd-widget-description", value:"Enjoy a beautiful daily photo selected by our community.", comment: "Text for description of Picture of the day widget displayed when adding to home screen.")
+    }
+
+}
+
+
+// MARK: - Widget
+
+struct PictureOfTheDayWidget: Widget {
+	private let kind: String = WidgetController.SupportedWidget.pictureOfTheDay.identifier
+
+	public var body: some WidgetConfiguration {
+		StaticConfiguration(kind: kind, provider: PictureOfTheDayProvider(), content: { entry in
+			PictureOfTheDayView(entry: entry)
+		})
+        .configurationDisplayName(PictureOfTheDayWidget.LocalizedStrings.pictureOfTheDayWidgetTitle)
+        .description(PictureOfTheDayWidget.LocalizedStrings.pictureOfTheDayWidgetDescription)
+		.supportedFamilies([.systemSmall, .systemMedium, .systemLarge])
+	}
+}
+
+// MARK: - Data
+
+/// A data source and operation helper for all Picture of the day widget data
+final class PictureOfTheDayData {
+
+	// MARK: Properties
+
+	static let shared = PictureOfTheDayData()
+
+	private var imageInfoFetcher = MWKImageInfoFetcher()
+	private var dataStore: MWKDataStore {
+		MWKDataStore.shared()
+	}
+
+	let sampleEntry = PictureOfTheDayEntry(date: Date(), image: #imageLiteral(resourceName: "PictureOfTheYear_2019"), imageDescription: "Two bulls running while the jockey holds on to them in pacu jawi (from Minangkabau, \"bull race\"), a traditional bull race in Tanah Datar, West Sumatra, Indonesia. 2015, Final-45.")
+	let placeholderEntry = PictureOfTheDayEntry(date: Date(), contentDate: nil, contentURL: nil, imageURL: nil, image: nil, imageDescription: nil)
+
+	// MARK: Public
+
+	func fetchLatestAvailablePictureEntry(usingImageCache: Bool = false, _ completion: @escaping (PictureOfTheDayEntry) -> Void) {
+		// We could Result type the completion, but it's not probably worth the complexity for the widget's use cases
+		guard let contentGroup = dataStore.viewContext.newestGroup(of: .pictureOfTheDay), let imageContent = contentGroup.contentPreview as? WMFFeedImage else {
```

Comment:
One thing I've wondered about when seeing `viewContext.newestGroup(of:)` is whether it'll scoop up the wrong (non-primary) language POTD content groups, simply because it's considered "newer" than the latest primary language content group. Doesn't seem like these managed object utilities prioritize their primary language but I could be missing it somewhere.

Reply:
I updated to `newestVisibleGroup`, per discussion that users intentionally may have hidden a particular card on the Explore feed and wouldn't want to see it again. That discussion also confirmed that this method already takes primary language into account.